### PR TITLE
[#105] Blocked connections metrics

### LINKED ISSUE #105: metrics: pgagroal_limit gauge
Would it be possible to add a value for number of waiting clients when pool limit is reached (for a given user/database) ?

### PR DESCRIPTION
**Pull request by fluca1978**
Creates a new metric named "connection awaiting" to keep track of the
number of connection enqueued but not served due to a reached limit.

This commit introduces a new field in the `prometheus` structure named
`connection_awaiting` to keep track of the connections that have been
suspended due to `blocking_timeout` configuration parameter.
Whenever a connection is suspended by the pool, the
`pgagroal_prometheus_connection_awaiting()` function is invoked to
increment the counter of awaiting connections, and following the
opposite happens with the
`pgagroal_prometheus_connection_unawaiting()` that decrements the
counter, since the connection workflow is going to restart (or abort).

The metric is exported as `pgagroal_connection_awaiting`.

Close #105

### CODE REVIEW

**Comment by jesperpedersen** [comment]
This is the general `blocking_timeout` metrics.

#105 is about having this information for each limit entry.

File: src/libpgagroal/prometheus.c [hunk 1]
```diff
@@ -1697,6 +1725,12 @@ pool_information(int client_fd)
    data = append_ulong(data, atomic_load(&prometheus->connection_success));
    data = append(data, "\n\n");
 
+   data = append(data, "#HELP pgagroal_connection_awaiting Number of connection awaiting\n");
+   data = append(data, "#TYPE pgagroal_connection_awaiting counter\n");
```
[the remark this answers is not in the document]
Comment (fluca1978) [reply src/libpgagroal/prometheus.c:1729]: Ok, fixing it.

**Comment by jesperpedersen** [comment]
Do you want the #105 functionality in this PR, or have a separate issue for it ?

**Comment by jesperpedersen** [comment]
You will need a list of `atomic_ulong` for each https://github.com/agroal/pgagroal/blob/master/src/include/pgagroal.h#L203 defined (`number_of_limits`). Increment/decrement based on the username/database the correct entry.

Then output a Prometheus map with the information like

```
pgagroal_limit_awaiting{user="u",database="d"} 1
```

See https://github.com/agroal/pgagroal/blob/master/src/libpgagroal/prometheus.c#L1427

**Comment by fluca1978** [comment]
I'm starting to develop on a different PR.

**Pull request closed without merging**